test/system/src/test/java/io/pravega/test/system/StreamCutsTest.java

```diff
@@ -0,0 +1,310 @@
+        log.debug("Finished creating StreamCuts.");
+
+        // Third, ensure that reader groups can correctly read slice by slice from different Streams.
+        readSliceBySliceAndVerify(readerGroupManager, clientFactory, streamSlices);
+
+        // Fourth, perform different combinations of StreamCuts and verify that read event boundaries are still correct.
+        combineSlicesAndVerify(readerGroupManager, clientFactory, streamSlices);
+
+        // Finally, Test that a reader group can be reset correctly.
+        ReaderGroupConfig firsSliceConfig = ReaderGroupConfig.builder()
+                                                             .stream(Stream.of(SCOPE, STREAM_ONE))
+                                                             .stream(Stream.of(SCOPE, STREAM_TWO))
+                                                             .startingStreamCuts(streamSlices.get(0))
+                                                             .build();
+        readerGroup.resetReaderGroup(firsSliceConfig);
+        log.info("Resetting existing reader group {} to stream cut {}.", READER_GROUP, streamSlices.get(0));
+        final int parallelSegments = RG_PARALLELISM_ONE + RG_PARALLELISM_TWO;
+        final int readEvents = readEventFutures(clientFactory, readerGroup.getGroupName(),
+                parallelSegments).stream().map(CompletableFuture::join).reduce((a, b) -> a + b).get();
+        assertEquals("Expected read events: ", TOTAL_EVENTS - CUT_SIZE, readEvents);
+        log.info("All events correctly read from StreamCut slices on multiple Streams. StreamCuts test passed.");
+    }
+
+    @After
+    public void tearDown() {
+        streamManager.close();
+        ExecutorServiceHelpers.shutdown(executor);
+    }
+
+    // Start utils region
+
+    /**
+     * This method performs slices in streams of non-consecutive StreamCuts. For instance, say that we generate 5 cuts
+     * in this order: C1, C2, C3, C4, C5. We then read slices of a stream formed like this:
+     *
+     * [C1, C3), [C1, C4), [C1, C5)
+     * [C2, C4), [C2, C5)
+     * [C3, C5)
+     *
+     * Note that all the consecutive slices have been previously tested, so we avoid them to shorten test execution.
+     * Moreover, take into account that a increase in the number of slices greatly lengthen execution time.
+     *
+     * @param manager Group manager for this scope.
+     * @param clientFactory Client factory to instantiate new readers.
+     * @param streamSlices StreamCuts lists to be combined and tested via bounded processing.
+     */
+    private void combineSlicesAndVerify(ReaderGroupManager manager, ClientFactory clientFactory,
+                                        List<Map<Stream, StreamCut>> streamSlices) {
+        ReaderGroupConfig.ReaderGroupConfigBuilder configBuilder = ReaderGroupConfig.builder()
+                                                                                    .stream(Stream.of(SCOPE, STREAM_ONE))
```

Comment:
can this readergroupconfig be  made global and defined in setup or some where , as it is used in 3 places.

Reply:
Done.